codeserver/ubi9-python-3.11/Dockerfile.konflux.cpu

```diff
@@ -0,0 +1,147 @@
+#    with default UID and GID (1001:0)
+# Supported combinations of UID:GID are thus following:
+# UID=1001 && GID=0
+# UID=<any>&& GID=0
+# UID=1001 && GID=<any>
+RUN sed -i -f ${NGINX_APP_ROOT}/nginxconf.sed ${NGINX_CONF_PATH} && \
+    mkdir -p ${NGINX_APP_ROOT}/etc/nginx.d/ && \
+    mkdir -p ${NGINX_APP_ROOT}/etc/nginx.default.d/ && \
+    mkdir -p ${NGINX_APP_ROOT}/api/ && \
+    mkdir -p ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    mkdir -p ${NGINX_LOG_PATH} && \
+    mkdir -p ${NGINX_PERL_MODULE_PATH} && \
+    chown -R 1001:0 ${NGINX_CONF_PATH} && \
+    chown -R 1001:0 ${NGINX_APP_ROOT}/etc && \
+    chown -R 1001:0 ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    chown -R 1001:0 /var/lib/nginx /var/log/nginx /run && \
+    chmod    ug+rw  ${NGINX_CONF_PATH} && \
+    chmod -R ug+rwX ${NGINX_APP_ROOT}/etc && \
+    chmod -R ug+rwX ${NGINX_CONTAINER_SCRIPTS_PATH}/nginx-start && \
+    chmod -R ug+rwX /var/lib/nginx /var/log/nginx /run && \
+    rpm-file-permissions && \
+    # Ensure the temporary directory and target directory have the correct permissions
+    mkdir -p /opt/app-root/src/.local/share/code-server/extensions && \
+    mkdir -p /opt/app-root/src/.local/share/code-server/coder-logs && \
+    chown -R 1001:0 /opt/app-root/src/.local/share/code-server && \
+    chown -R 1001:0 /opt/app-root/extensions-temp && \
+    chown -R 1001:0 /opt/app-root/src/.config/code-server
+
+## Configure nginx
+COPY ${CODESERVER_SOURCE_CODE}/nginx/serverconf/ /opt/app-root/etc/nginx.default.d/
+COPY ${CODESERVER_SOURCE_CODE}/nginx/httpconf/ /opt/app-root/etc/nginx.d/
+COPY ${CODESERVER_SOURCE_CODE}/nginx/api/ /opt/app-root/api/
+
+# Launcher
+COPY --chown=1001:0 ${CODESERVER_SOURCE_CODE}/run-code-server.sh ${CODESERVER_SOURCE_CODE}/run-nginx.sh ./
+
+ENV SHELL=/bin/bash
+
+ENV PYTHONPATH=/opt/app-root/bin/python3
+
+USER 1001
+
+# Install useful packages from Pipfile.lock
+COPY ${CODESERVER_SOURCE_CODE}/Pipfile.lock ./
+
+# Install packages and cleanup
+RUN echo "Installing softwares and packages" && \
+    micropipenv install && \
+    rm -f ./Pipfile.lock && \
+    # Fix permissions to support pip in Openshift environments \
+    chmod -R g+w /opt/app-root/lib/python3.11/site-packages && \
+    fix-permissions /opt/app-root -P
+
+WORKDIR /opt/app-root/src
+
+CMD ["/opt/app-root/bin/run-code-server.sh"]
+
+LABEL name="rhoai/odh-workbench-codeserver-datascience-cpu-py311-rhel9" \
+      com.redhat.component="odh-workbench-codeserver-datascience-cpu-py311-rhel9" \
+      io.k8s.display-name="odh-workbench-codeserver-datascience-cpu-py311-rhel9" \
+      summary="code-server image with python 3.11 based on UBI 9" \
+      description="code-server image with python 3.11 based on UBI9" \
+      io.k8s.description="code-server image with python 3.11 based on UBI9" \
+      com.redhat.license_terms="https://www.redhat.com/licenses/Red_Hat_Standard_EULA_20191108.pdf"
```

Comment:
> A label key is the left-hand side of the key-value pair. Keys are alphanumeric strings which may contain periods (.), underscores (_), slashes (/), and hyphens (-).
https://docs.docker.com/engine/manage-resources/labels/

hadolint complains about the underscore, but even according to docs, underscore is fine... hadolint is wrong?